packages/graph/hash_graph/lib/graph/src/types/schema/combinator.rs

```diff
@@ -0,0 +1,157 @@
+use serde::{Deserialize, Serialize};
+
+use crate::types::schema::{array::TypedArray, ValidationError};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Optional<T> {
+    None {},
```

Comment:
Is this explicitly an empty struct because of implications by `serde(untagged)`?

Reply:
Yes. If it would be `None`, the serialized field would be `null`. But with `None {}` the serialized field is `{}`